src/main/kotlin/nl/info/zac/search/converter/DocumentZoekObjectConverter.kt

```diff
@@ -82,12 +82,16 @@ class DocumentZoekObjectConverter @Inject constructor(
                 ondertekeningDatum = convertToDate(ondertekening.datum)
                 setIndicatie(DocumentIndicatie.ONDERTEKEND, true)
             }
-            setIndicatie(DocumentIndicatie.VERGRENDELD, informatieobject.locked)
-            setIndicatie(DocumentIndicatie.GEBRUIKSRECHT, informatieobject.indicatieGebruiksrecht)
-            setIndicatie(
-                DocumentIndicatie.BESLUIT,
-                brcClientService.isInformatieObjectGekoppeldAanBesluit(informatieobject.url)
-            )
+            informatieobject.locked?.let { setIndicatie(DocumentIndicatie.VERGRENDELD, it) }
```

Comment:
Does this mean that if there is no current locked state, this will never be set? BUT if `locked` is false, the state will also be set to `VERGRENDELD` ? Or is that handled in the `setIndicatie` method - it doesn't look like that from here...

Would you not want to stick to the previous construction and either deal with a tri-state (true, false, null) in the `setIndicatie` method or force it to be `false` by default, per something like this?

```suggestion
setIndicatie(DocumentIndicatie.VERGRENDELD, informatieobject.locked ?: false)
```

Reply:
Also I - just me maybe - have a slight dislike for the use of `let` in this situation, as I'd expect the method inside to act upon the `it` context. I think I'd prefer `run` here instead, as that indicates to me that `it` is not necessarily the subject of change.

Reply:
Good question. It's a bit of guesswork to decide how to do null handling now really. The way it used to work was: throw a nullpointer when `informatieobject.locked` is null and everything breaks. Now it's up to us to decide how to handle it. First question is: can it even ever be null according to the ZGW API? I would need to check.

Reply:
I agree, according to the ZGW API specs `locked` can never be null. Same with `url`. So I reverted those changes.

As to using `let` vs `run`, I think `let` is the better choice here. Also see: https://medium.com/@fatihcoskun/kotlin-scoping-functions-apply-vs-with-let-also-run-816e4efb75f5

>  Use the let() function in either of the following cases:
> 
> execute code if a given value is not null
> convert a nullable object to another nullable object
> limit the scope of a single local variable